cabal-install/src/Distribution/Client/ScriptUtils.hs

```diff
@@ -0,0 +1,342 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+
+-- | Utilities to help commands with scripts
+--
+module Distribution.Client.ScriptUtils (
+    getScriptCacheDirectoryRoot, getScriptHash, getScriptCacheDirectory, ensureScriptCacheDirectory,
+    withContextAndSelectors, AcceptNoTargets(..), TargetContext(..),
+    updateContextAndWriteProjectFile, updateContextAndWriteScriptProjectFiles,
+    fakeProjectSourcePackage, lSrcpkgDescription
+  ) where
+
+import Prelude ()
+import Distribution.Client.Compat.Prelude hiding (toList)
+
+import Distribution.Compat.Lens
+import qualified Distribution.Types.Lens as L
+
+import Distribution.Client.ProjectOrchestration
+import Distribution.Client.DistDirLayout
+         ( DistDirLayout(..) )
+import Distribution.Client.NixStyleOptions
+         ( NixStyleFlags (..) )
+import Distribution.Client.Setup
+         ( GlobalFlags(..), ConfigFlags(..) )
+import Distribution.Client.Config
+         ( getCabalDir )
+import Distribution.Simple.Flag
+         ( fromFlagOrDefault )
+import Distribution.Simple.Setup
+         ( Flag(..) )
+import Distribution.CabalSpecVersion (CabalSpecVersion (..), cabalSpecLatest)
```

Comment:
Probably format this line as well, just like the others?

Reply:
A lot of the code for ScriptUtils came from CmdRun, but since it's not being explicitly diffed with CmdRun, I guess there's no reason to keep it.